src/main/java/de/caritas/cob/userservice/api/service/session/ConsultantSessionTopicEnrichmentService.java

```diff
@@ -0,0 +1,85 @@
+package de.caritas.cob.userservice.api.service.session;
+
+import static java.util.Objects.nonNull;
+
+import de.caritas.cob.userservice.api.adapters.web.dto.ConsultantSessionDTO;
+import de.caritas.cob.userservice.api.adapters.web.dto.SessionTopicDTO;
+import de.caritas.cob.userservice.api.service.consultingtype.TopicService;
+import de.caritas.cob.userservice.topicservice.generated.web.model.TopicDTO;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@ConditionalOnExpression("${feature.topics.enabled:true}")
+@Slf4j
+public class ConsultantSessionTopicEnrichmentService {
+
+  private final @NonNull TopicService topicService;
+
+  public ConsultantSessionDTO enrichSessionWithTopicData(ConsultantSessionDTO session) {
+    if (session.getMainTopic() != null && session.getMainTopic().getId() != null
```

Comment:
according to clean code, please extract the content of the if condition to a separate method like shouldEnrichTopics